esphome/components/madoka/madoka.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include <queue>
+#include <map>
+
+#include "esphome/core/component.h"
+#include "esphome/components/ble_client/ble_client.h"
+#include "esphome/components/esp32_ble_tracker/esp32_ble_tracker.h"
+#include "esphome/components/climate/climate.h"
+
+// #define USE_ESP32
+
+#ifdef USE_ESP32
+
+#include <esp_gattc_api.h>
+
+static const uint8_t MAX_CHUNK_SIZE = 20;
+static const uint8_t BLE_SEND_MAX_RETRIES = 5;
+
+namespace esphome {
+namespace madoka {
+
+static const char *const TAG = "madoka";
+
+using chunk = std::vector<uint8_t>;
+using message = std::vector<uint8_t>;
+
+struct Setpoint {
+  uint16_t cooling;
+  uint16_t heating;
+};
+
+struct FanSpeed {
+  uint8_t cooling;
+  uint8_t heating;
+};
+
+struct SensorReading {
+  uint8_t indoor;
+  uint8_t outdoor;
+};
+
+struct Status {
+  bool status;
+  uint8_t mode;
+};
+
+namespace espbt = esphome::esp32_ble_tracker;
+
+#define TO_ESPBTUUID(x) espbt::ESPBTUUID::from_raw(std::string(x))
+
+#define MADOKA_SERVICE_UUID TO_ESPBTUUID("2141e110-213a-11e6-b67b-9e71128cae77")
```

Comment:
Please avoid the defines here
```suggestion
static const espbt::ESPBTUUID MADOKA_SERVICE_UUID = espbt::ESPBTUUID::from_raw("2141e110-213a-11e6-b67b-9e71128cae77");
```